### PR TITLE
update to 3.21.7

### PR DESCRIPTION
**Pull request by jezdez**

### CODE REVIEW

**Review by chenghlee — APPROVED** [review]

**Comment by jezdez** [comment]
@chenghlee Can we restaret prefect again?

**Comment by skupr-anaconda** [comment]
- `conda  >=4.5`: too much spaces between package and version'
- Missing `wheel` package in requirements/host
- License should be `BSD-3-Clause`
- `conda-package-handling >=1.3` and `glob2 >=0.6` aren't mentioned in the upstream source https://github.com/conda/conda-build/blob/b98d7ec0c10e8b933f4663abe47b298f3fe08f8d/setup.py#L11
- Don't we use tests https://github.com/AnacondaRecipes/conda-build-feedstock/blob/c4c02fee7149ccc2a2b4165324430a424c81bc6b/recipe/meta.yaml#L61
- Does it make sense add other imports: `conda_build.cli`,  `conda_build.os_utils`, `conda_build.skeletons` ?https://github.com/AnacondaRecipes/conda-build-feedstock/blob/c4c02fee7149ccc2a2b4165324430a424c81bc6b/recipe/meta.yaml#L71

**Review by skupr-anaconda — APPROVED** [review]

**Comment by jezdez** [comment]
> * `conda  >=4.5`: too much spaces between package and version'
> * Missing `wheel` package in requirements/host
> * License should be `BSD-3-Clause`

All done.

> * `conda-package-handling >=1.3` and `glob2 >=0.6` aren't mentioned in the upstream source https://github.com/conda/conda-build/blob/b98d7ec0c10e8b933f4663abe47b298f3fe08f8d/setup.py#L11

Opened https://github.com/conda/conda-build/pull/4340 because they are missing there.

> * Don't we use tests? https://github.com/AnacondaRecipes/conda-build-feedstock/blob/c4c02fee7149ccc2a2b4165324430a424c81bc6b/recipe/meta.yaml#L61

I'm not sure, @chenghlee, given that conda-build runs a LONG time (>2 hours) right now, I'm inclined to skip this until we have a better test system in place.

@skupr-anaconda Are these lines used automatically in any way? Or is this for you to better review if all is good?

> * Does it make sense to add other imports: `conda_build.cli`,  `conda_build.os_utils`, `conda_build.skeletons` ?https://github.com/AnacondaRecipes/conda-build-feedstock/blob/c4c02fee7149ccc2a2b4165324430a424c81bc6b/recipe/meta.yaml#L71

All those modules are empty, but maybe I'm missing something, would adding those packages test imports in the subpackages as well, e.g. `conda_build.cli.main_build`?

**Comment by skupr-anaconda** [comment]
Yeah, `conda_build.cli.main_build` is a good example to add in test/imports

**Comment by jezdez** [comment]
@skupr-anaconda Could you help me with taking a look at the build failure on the s390x build? I don't have access to this.

**Comment by skupr-anaconda** [comment]
@dbast on s390x:
```
Task 'build': Exception encountered during task execution!
Traceback (most recent call last):
  File "/home/prefect/miniconda3/envs/prefect/lib/python3.9/site-packages/prefect/client/secrets.py", line 137, in get
    value = secrets[self.name]
KeyError: 'GITHUB_CLONE_AND_STATUS'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/prefect/miniconda3/envs/prefect/lib/python3.9/site-packages/prefect/client/secrets.py", line 145, in get
    result = self.client.graphql(
  File "/home/prefect/miniconda3/envs/prefect/lib/python3.9/site-packages/prefect/client/client.py", line 569, in graphql
    raise ClientError(result["errors"])
prefect.exceptions.ClientError: [{'path': ['secret_value'], 'message': 'No value found for the requested key ("GITHUB_CLONE_AND_STATUS") in tenant 5436cd20-42e7-4c53-abdc-af9e7f14c49c', 'extensions': {'code': 'INTERNAL_SERVER_ERROR'}}]

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/prefect/miniconda3/envs/prefect/lib/python3.9/site-packages/prefect/engine/task_runner.py", line 859, in get_task_run_state
    value = prefect.utilities.executors.run_task_with_timeout(
  File "/home/prefect/miniconda3/envs/prefect/lib/python3.9/site-packages/prefect/utilities/executors.py", line 518, in run_task_with_timeout
    return fut.result(timeout=task.timeout)
  File "/home/prefect/miniconda3/envs/prefect/lib/python3.9/concurrent/futures/_base.py", line 445, in result
    return self.__get_result()
  File "/home/prefect/miniconda3/envs/prefect/lib/python3.9/concurrent/futures/_base.py", line 390, in __get_result
    raise self._exception
  File "/home/prefect/miniconda3/envs/prefect/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/prefect/miniconda3/envs/prefect/lib/python3.9/site-packages/prefect/utilities/executors.py", line 512, in run_with_ctx
    return task.run(*args, **kwargs)  # type: ignore
  File "<string>", line 307, in build
  File "<string>", line 96, in report_state
  File "/home/prefect/prefect/flows/commons.py", line 254, in github_status_report
    github_token = Secret("GITHUB_CLONE_AND_STATUS").get()  # noqa
  File "/home/prefect/miniconda3/envs/prefect/lib/python3.9/site-packages/prefect/client/secrets.py", line 155, in get
    raise KeyError(
KeyError: 'The secret GITHUB_CLONE_AND_STATUS was not found.  Please ensure that it was set correctly in your tenant: https://docs.prefect.io/orchestration/concepts/secrets.html'
```

**Comment by dbast** [comment]
Retriggered on s390x (Need to catch the exception in case of status update issues and retry).

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,34 +28,31 @@ requirements:
...
-    - enum34             # [py27]
+    - conda >=4.5
+    - conda-package-handling >=1.3
     - filelock
-    - futures            # [py27]
     - jinja2
     - patchelf      # [linux]
     - pkginfo
     - psutil
+    - py-lief  # [not win]
```
Comment (isuruf) [inline recipe/meta.yaml:42]: why this change?

Comment (chenghlee) [reply recipe/meta.yaml:42]: Bug; see AnacondaRecipes/conda-build-feedstock#11
